lib/style/deprecations.ex

```diff
@@ -28,56 +28,85 @@ defmodule Styler.Style.Deprecations do
   defp style({{:., dm, [{_, _, [:Path]} = mod, :safe_relative_to]}, funm, args}),
     do: {{:., dm, [mod, :safe_relative]}, funm, args}
 
+  # Pipe version for:
   # Path.safe_relative_to/2 => Path.safe_relative/2
-  # Path.safe_relative/2 is available since v1.14
-  # TODO: Remove after Elixir v1.19
   defp style({:|>, m, [lhs, {{:., dm, [{:__aliases__, am, [:Path]}, :safe_relative_to]}, funm, args}]}),
     do: {:|>, m, [lhs, {{:., dm, [{:__aliases__, am, [:Path]}, :safe_relative]}, funm, args}]}
 
-  if Version.match?(System.version(), ">= 1.16.0-dev") do
-    # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
-    defp style({{:., _, [{_, _, [:File]}, :stream!]} = f, fm, [path, {:__block__, _, [modes]} = opts, lob]})
-         when is_list(modes),
-         do: {f, fm, [path, lob, opts]}
-
-    # For ranges where `start > stop`, you need to explicitly include the step
-    # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
-    # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
-    defp style({{:., _, [{_, _, [module]}, :slice]} = f, funm, [enumerable, {:.., _, [_, _]} = range]})
-         when module in [:Enum, :String],
-         do: {f, funm, [enumerable, add_step_to_decreasing_range(range)]}
-
-    # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
-    defp style({:|>, m, [lhs, {{_, _, [{_, _, [:File]}, :stream!]} = f, fm, [{:__block__, _, [modes]} = opts, lob]}]})
-         when is_list(modes),
-         do: {:|>, m, [lhs, {f, fm, [lob, opts]}]}
-
-    # For ranges where `start > stop`, you need to explicitly include the step
-    # Enum.slice(enumerable, 1..-2) => Enum.slice(enumerable, 1..-2//1)
-    # String.slice("elixir", 2..-1) => String.slice("elixir", 2..-1//1)
-    defp style({:|>, m, [lhs, {{:., _, [{_, _, [mod]}, :slice]} = f, funm, [{:.., _, [_, _]} = range]}]})
-         when mod in [:Enum, :String],
-         do: {:|>, m, [lhs, {f, funm, [add_step_to_decreasing_range(range)]}]}
+  # File.stream!(file, options, line_or_bytes) => File.stream!(file, line_or_bytes, options)
```

Comment:
`File.stream!/3` only had its args switched in 1.16 - i looked into this when you first PR'd hoping we could skip the conditional compile but no dice

here's 1.15.7
```
iex(3)> "mix.exs" |> File.stream!(:line, [:raw]) |> Enum.to_list
** (FunctionClauseError) no function clause matching in File.normalize_modes/2

    The following arguments were given to File.normalize_modes/2:

        # 1
        :line

        # 2
        true

    Attempted function clauses (showing 6 out of 6):

        defp normalize_modes([:utf8 | rest], binary?)
        defp normalize_modes([:read_ahead | rest], binary?)
        defp normalize_modes([mode | rest], _binary?) when mode === :charlist or mode === :char_list
        defp normalize_modes([mode | rest], binary?)
        defp normalize_modes([], true)
        defp normalize_modes([], false)

    (elixir 1.15.7) lib/file.ex:1854: File.normalize_modes/2
    (elixir 1.15.7) lib/file.ex:1730: File.stream!/3
    iex:3: (file)
```

Reply:
likely the sigil and slice functions are fine to always rewrite though.

Reply:
@novaugust oops, thanks for double-checking, that `File.stream!` fell through the cracks.

Reply:
Added back the conditional for `File.stream!`, please let me know what you think.